packages/react/src/auto/shadcn/inputs/ShadcnAutoNumberInput.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import type { Control } from "../../../useActionForm.js";
+import { autoInput } from "../../AutoInput.js";
+import { useStringInputController } from "../../hooks/useStringInputController.js";
+import type { ShadcnElements } from "../elements.js";
+import { makeShadcnAutoTextInput } from "./ShadcnAutoTextInput.js";
+
+export const makeShadcnAutoNumberInput = ({ Input, Label }: Pick<ShadcnElements, "Input" | "Label">) => {
+  const ShadcnAutoTextInput = makeShadcnAutoTextInput({ Input, Label });
+
+  return autoInput((props: { field: string; control?: Control<any>; className?: string }) => {
+    const { field, control } = props;
+    const { metadata, value } = useStringInputController({ field, control });
+
+    const step =
+      metadata.configuration.__typename === "GadgetNumberConfig" && metadata.configuration.decimals && metadata.configuration.decimals > 0
+        ? getStepFromNumberOfDecimals(metadata.configuration.decimals)
+        : value
+        ? getStepFromNumberOfDecimals(countNumberOfDecimals(`${value}`))
+        : 1;
+
+    return <ShadcnAutoTextInput {...props} step={step} type={"number"} />;
+  });
+};
+
+// TODO: move to a shared location
```

Comment:
👍 definitely -- as well as the stuff above with the `step` if that is shared with polaris